api/observability/v1/input_types.go

```diff
@@ -0,0 +1,294 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// InputType specifies the type of log input to create.
+//
+// +kubebuilder:validation:Enum:=audit;application;infrastructure;receiver
+type InputType string
+
+// Reserved input names.
+const (
+	// InputTypeApplication contains all the non-infrastructure container logs.
+	InputTypeApplication InputType = "application"
+	// InputTypeInfrastructure contains infrastructure containers and system logs.
+	InputTypeInfrastructure InputType = "infrastructure"
+	// InputTypeAudit contains system audit logs.
+	InputTypeAudit InputType = "audit"
+	// InputTypeReceiver defines a network receiver for receiving logs from non-cluster sources.
+	InputTypeReceiver InputType = "receiver"
+)
+
+var (
+	InputTypes = []InputType{
+		InputTypeApplication,
+		InputTypeInfrastructure,
+		InputTypeAudit,
+		InputTypeReceiver,
+	}
+)
+
+// InputSpec defines a selector of log messages for a given log type.
+type InputSpec struct {
+	// Name used to refer to the input of a `pipeline`.
```

Comment:
Kube conventions suggest using the json tag name for the start of the comment, since users are used to seeing serialized versions of fields, not the Go versions